arithmetization/src/main/java/net/consensys/linea/zktracer/module/stp/Stp.java

```diff
@@ -0,0 +1,424 @@
+        this.wcp.callLT(longToBytes32(chunk.gasActual()), longToBytes32(chunk.gasPrelim()));
+        if (!chunk.oogx()) {
+          this.mod.callDiv(longToBytes32(getGDiff(chunk)), longToBytes32(64L));
+        }
+      }
+      case CALL, CALLCODE, DELEGATECALL, STATICCALL -> {
+        final StpChunk chunk = getCallData(frame);
+        this.chunks.add(chunk);
+        this.wcp.callLT(longToBytes32(chunk.gasActual()), Bytes32.ZERO);
+        if (cctv(chunk.opCode())) {
+          this.wcp.callISZERO(Bytes32.leftPad(chunk.value()));
+        }
+        this.wcp.callLT(longToBytes32(chunk.gasActual()), longToBytes32(chunk.gasPrelim()));
+        if (!chunk.oogx()) {
+          this.mod.callDiv(longToBytes32(getGDiff(chunk)), longToBytes32(64L));
+          this.wcp.callLT(chunk.gas().orElseThrow(), longToBytes32(get63of64GDiff(chunk)));
+        }
+      }
+    }
+  }
+
+  private StpChunk getCreateData(final MessageFrame frame) {
+    final Address to = getDeploymentAddress(frame);
+    final Long gasRemaining = frame.getRemainingGas();
+    final Long gasMxp = getGasMxpCreate(frame);
+    final Long gasPrelim = GasConstants.G_CREATE.cost() + gasMxp;
+    return new StpChunk(
+        OpCode.of(frame.getCurrentOperation().getOpcode()),
+        gasRemaining,
+        gasPrelim,
+        gasRemaining < gasPrelim,
+        gasMxp,
+        frame.getWorldUpdater().get(frame.getContractAddress()).getBalance(),
+        to,
+        Bytes32.leftPad(frame.getStackItem(0)));
+  }
+
+  private StpChunk getCallData(final MessageFrame frame) {
+    final OpCode opcode = OpCode.of(frame.getCurrentOperation().getOpcode());
+    final long gasActual = frame.getRemainingGas();
+    final Bytes32 value = cctv(opcode) ? Bytes32.leftPad(frame.getStackItem(2)) : Bytes32.ZERO;
+    final Address to = Words.toAddress(frame.getStackItem(1));
+    final Long gasMxp = getGasMxpCall(frame);
+    final boolean toWarm = frame.isAddressWarm(to);
+    final boolean toExists =
+        opcode == OpCode.CALLCODE
+            || (frame.getWorldUpdater().get(to) != null
+                && !frame.getWorldUpdater().get(to).isEmpty());
+
+    long gasPrelim = gasMxp;
+    if (!value.isZero() && cctv(opcode)) {
+      gasPrelim += GasConstants.G_CALL_VALUE.cost();
+    }
+    if (toWarm) {
+      gasPrelim += GasConstants.G_WARM_ACCESS.cost();
+    } else {
+      gasPrelim += GasConstants.G_COLD_ACCOUNT_ACCESS.cost();
+    }
+    if (!toExists) {
+      gasPrelim += GasConstants.G_NEW_ACCOUNT.cost();
+    }
+    final boolean oogx = gasActual < gasPrelim;
+    return new StpChunk(
+        opcode,
+        gasActual,
+        gasPrelim,
+        oogx,
+        gasMxp,
+        frame.getWorldUpdater().get(frame.getContractAddress()).getBalance(),
+        to,
+        value,
+        toExists,
+        toWarm,
+        Bytes32.leftPad(frame.getStackItem(0)));
+  }
+
+  private boolean cctv(OpCode opCode) {
```

Comment:
cctv = « coucou tu veux voir [ma bite] » ? “China Central TeleVision”?

Reply:
call can transfer value, directly from the spec. It goes with @OlivierBBB's SUX.